pkg/querier/http.go

```diff
@@ -58,44 +59,43 @@ func directionParam(values url.Values, name string, def logproto.Direction) (log
 	return logproto.Direction(d), nil
 }
 
-// QueryHandler is a http.HandlerFunc for queries.
-func (q *Querier) QueryHandler(w http.ResponseWriter, r *http.Request) {
-	params := r.URL.Query()
-	query := params.Get("query")
+func httpRequestToQueryRequest(httpRequest *http.Request) (logproto.QueryRequest, error) {
+	params := httpRequest.URL.Query()
+	now := time.Now()
+	queryRequest := logproto.QueryRequest{
+		Regex: params.Get("regexp"),
+	}
+
+	queryRequest.Query = params.Get("query")
 	limit, err := intParam(params, "limit", defaultQueryLimit)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
+		return queryRequest, err
```

Comment:
Also in this change we're loosing the ability to return meaningful status codes.  Can you use `httpgrpc.Errorf()` which allows you to return error codes?

Reply:
Actually websocket and normal http requests use different status codes.
Here are the websocket error codes:
https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent#Properties
Maybe I should map common error codes from http to websocket error codes?

Reply:
I wont need conversion between error codes since I am doing [this](https://github.com/grafana/loki/pull/439#discussion_r271600361)